tests/nested/manual/core20-cloud-init-maas-signed-seed-data/task.yaml

```diff
@@ -139,53 +170,117 @@ execute: |
   . "$TESTSLIB/nested.sh"
 
   echo "Waiting for snapd to react to cloud-init"
-  seen=0
-  # the first message is about snapd noticing that cloud-init is in error, and
-  # for how long snapd will wait for cloud-init, this message should show up
-  # relatively quickly
-  for i in $(seq 60); do
-      if tests.nested exec "sudo journalctl --no-pager -u snapd" | MATCH "cloud-init reported to be in error state, will disable in 3 minutes"; then
-          seen=1
-          break
-      fi
-      sleep 1
-  done
-  if [ "$seen" = "0" ]; then
-      echo "snapd did not notice cloud-init in error state, test failed"
-      exit 1
-  fi
-  seen=0
-
-  # we now wait for 3 minutes before snapd gives up waiting and disables 
-  # cloud-init
-  for i in $(seq 200); do
-      if tests.nested exec "sudo journalctl --no-pager -u snapd" | MATCH "reported to be in error state after 3 minutes"; then
-          seen=1
-          break
-      fi
-      sleep 1
-  done
-
-  if [ "$seen" = "0" ]; then
-      echo "snapd did not disable cloud-init, test failed"
-      exit 1
+
+  # there are two options here, one is where we actually let cloud-init stay 
+  # enabled with the variant where the gadget allowed it, in which case 
+  # cloud-init will be in error, and we actually have to check for two messages,
+  # and the other is where cloud-init is not allowed to run at all by disabling
+  # all datasources in which case snapd notices cloud-init as being disabled
+  if [ "$GADGET_DATASOURCE" = "explicitly-none" ]; then
+    seen=0
+
+    for i in $(seq 60); do
+        if tests.nested exec "sudo journalctl --no-pager -u snapd" | MATCH "cloud-init reported to be in disabled state"; then
+            seen=1
+            break
+        fi
+        sleep 1
+    done
+    if [ "$seen" = "0" ]; then
+        echo "snapd did not notice cloud-init in disabled state, test failed"
+        exit 1
+    fi
+
+    tests.nested exec "cloud-init status" | MATCH "status: disabled"
+
+  else
+    seen=0
+
+    # the first message is about snapd noticing that cloud-init is in error, and
+    # for how long snapd will wait for cloud-init, this message should show up
+    # relatively quickly
+    for i in $(seq 60); do
+        if tests.nested exec "sudo journalctl --no-pager -u snapd" | MATCH "cloud-init reported to be in error state, will disable in 3 minutes"; then
+            seen=1
+            break
+        fi
+        sleep 1
+    done
+    if [ "$seen" = "0" ]; then
+        echo "snapd did not notice cloud-init in error state, test failed"
+        exit 1
+    fi
+    seen=0
+
+    # we now wait for 3 minutes before snapd gives up waiting and disables 
+    # cloud-init
+    for i in $(seq 200); do
```

Comment:
same question about using retry command on the nested host

Reply:
done, let's see if it works